### ISSUE TITLE
Switching from Practice to Race

### ISSUE THREAD

**Issue by pminnieur** [apihlaja/node-irsdk#5]
I recently discovered that my Electron app starts hanging (doesn't respond anymore) when I switch for Practice to Race in game (it doesn't happend when I close iRacing and click on "Join" in the browser to join the race). I also don't get any error messages, the IRSDK seems to get a hiccup.

Since my software hangs, I have no debug trace or something alike. Its also hard to test because you have to join a Race and risk losing SR/IR if you only join to test that behavior :-/

**Comment by apihlaja** [comment]
Does it happen every time? Or only occasionally?  And what do you mean by hanging? node-irsdk part stops working or your js-app too? Just wondering if electron handles crashing native module somehow differently than node.js normally. node.js simply shuts down with some undescriptive exit code.

Switching session and exit+join *should* behave similarly. In both cases, sim sets `connected` bit to 0 and then back to 1 when its up. And node-irsdk reacts both exactly same: runs its internal shutdown and startup.

node-irsdk used to crash occasionally when loading new session or switching sessions. It haven´t happened for me lately, I thought I fixed it sometime ago but I never figured out how to reproduce it so I can´t be sure. If it happens always when switching sessions, I can test it in unofficial series. If wondering why I have so many PickUp Cup starts... ;-)

**Comment by pminnieur** [comment]
Two times my electron app become "unresponsive", meaning the window blanked out and Windows wanted to wait for it. I couldn't recover any data from that window/process. Another time, it looked like my app does no longer receive any updates from the game (nothing happend, but responsive window).

It may have been an error on my side of the application, again, it's very hard to debug that issue. I discovered one thing that may have caused the "no more updates" issue was my tick-counter (see https://github.com/pminnieur/irdash/blob/master/app/modules/iracing.js).

To not flood my app with data, I checked that the ingame ticks of updates are greater than the last one received. I guess iRacing will reset the tick counter when you go from practice to race, but no Disconnected/Connected event will happen (which reset my tick state). The new updates may start with 0 again and so my app would block updates until the tick counter is high enough again. I changed it to check only if its not equal to the current state, which may fixed the issue.

I'll keep an eye on it, but I guess we can close this issue for now. Sorry for that false positive ;-)
